### PR TITLE
[NumPy] Add NumPy support for triu

### PR DESCRIPTION
**Pull request by vexilligera**
## Description ##
This is directly modified from np.tril

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/api/operator/numpy/np_triu_op.cc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  using namespace runtime;
+  op::TriuParam param;
+  nnvm::NodeAttrs attrs;
+  const nnvm::Op* op = Op::Get("_npi_triu");
+  // inputs
+  param.k = args[1].operator int();
+  NDArray* inputs[] = {args[0].operator NDArray*()};
+
+  attrs.op = op;
+  attrs.parsed = std::move(param);
```
Comment (hzfan) [inline src/api/operator/numpy/np_triu_op.cc]: `attrs.parsed = param` to fix the clang-tidy error.

**Comment by haojin2** [comment]
@mxnet-bot run ci [all]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [all]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [centos-cpu]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [centos-cpu]

**Pull request merged**